doc/test-controller.md

```diff
@@ -0,0 +1,184 @@
+follow that guide for persistent hugepages & kernel module setup.
+
+Or, for ad-hoc:
+
+- Ensure at least 512 2MB hugepages.
+
+  ```bash
+  echo 512 | sudo tee  /sys/kernel/mm/hugepages/hugepages-2048kB/nr_hugepages
+  ```
+
+- Ensure several kernel modules are installed:
+
+  ```bash
+  modprobe xfs nvme_fabrics nvme_tcp nvme_rdma
+  ```
+
+- Ensure docker is installed and the service is running (OS specific)
+
+## Table of Contents
+
+- [Table of Contents](#table-of-contents)
+- [Local Docker Playground](#local-docker-playground)
+  - [Deploying](#deploying)
+- [Running the test suites](#running-the-test-suites)
+  - [Unit/Integration/Docs](#unitintegrationdocs)
+  - [BDD](#bdd)
+  - [Testing with a custom io-engine](#testing-with-a-custom-io-engine)
+- [Local K8s Playground](#local-k8s-playground)
+  - [Example](#example)
+
+## Local Docker Playground
+
+The Mayastor integration tests leverage docker in order to create a "cluster" with multiple components running as their own docker container within the same network.
+Specifically, the control-plane integration tests make use of the [deployer](https://github.com/openebs/mayastor-control-plane/blob/HEAD/deployer/README.md) which can setup these "clusters" for you, along with a very extensive range of options.
+
+### Deploying
+
+Starting a deployer "cluster", is then very simple:
+
+```console
+deployer start -s -i 2 -w 5s
+[/core] [10.1.0.3] /home/tiago/git/mayastor/controller/target/debug/core --store etcd.cluster:2379
+[/etcd] [10.1.0.2] /nix/store/7fvflmxl9a8hfznsc1sddp5az1gjlavf-etcd-3.5.13/bin/etcd --data-dir /tmp/etcd-data --advertise-client-urls http://[::]:2379 --listen-client-urls http://[::]:2379 --heartbeat-interval=1 --election-timeout=5
+[/io-engine-1] [10.1.0.5] /bin/io-engine -N io-engine-1 -g 10.1.0.5:10124 -R https://core:50051 --api-versions V1 -r /host/tmp/io-engine-1.sock --ptpl-dir /host/tmp/ptpl/io-engine-1 -p etcd.cluster:2379
+[/io-engine-2] [10.1.0.6] /bin/io-engine -N io-engine-2 -g 10.1.0.6:10124 -R https://core:50051 --api-versions V1 -r /host/tmp/io-engine-2.sock --ptpl-dir /host/tmp/ptpl/io-engine-2 -p etcd.cluster:2379
+[/rest] [10.1.0.4] /home/tiago/git/mayastor/controller/target/debug/rest --dummy-certificates --https rest:8080 --http rest:8081 --workers=1 --no-auth
+```
+
+> **NOTE**: Use `--io-engine-isolate` to given each engine a different cpu core
+> **NOTE**: Use `--developer-delayed` for sleep delay on each engine, reducing cpu usage
+> **NOTE**: For all options, check `deployer start --help`
+
+And with this we have a dual io-engine cluster which we can interact with.
+
+```console
+rest-plugin get nodes
+ ID           GRPC ENDPOINT   STATUS  VERSION
+ io-engine-2  10.1.0.6:10124  Online  v1.0.0-997-g17488f4a7da3
+ io-engine-1  10.1.0.5:10124  Online  v1.0.0-997-g17488f4a7da3
+```
+
+You can also use the swagger-ui available on the [localhost:8081](http://localhost:8081/v0/swagger-ui#).
```

Comment:
We can mention here that port forwarding may be required to use swagger UI.

Reply:
Are you thinking of K8s?

Reply:
I guess it depends where deployer is started. e.g typically if started in a VM that we ssh into, then have to port forward between local machine and VM. e.g
`ssh -L 8081:127.0.0.1:8081 username@host`

Reply:
If someone is starting it on a remote shell, then they should know they have to port-forward that, it won't magically appear on their hosts :)